### PR TITLE
write repodata in subprocesses

### LINKED ISSUE #33: Delete tqdm
Code has inconsistent verbose logging / tqdm usage, messing up the progress bars. Would be simpler to get rid of tqdm altogether.

Maybe use rich progress bars?

### PR DESCRIPTION
**Pull request by dholth**
~5m -> 1m20s. Also fix #33 , is any tqdm left?

### CODE REVIEW

File: conda_index/index/__init__.py [hunk 1]
```diff
@@ -540,74 +547,79 @@ def extract_wrapper(args: tuple):
...
+                                subdir=subdir,
+                                verbose=verbose,
+                                progress=progress,
+                                patch_generator=patch_generator,
+                                current_index_versions=current_index_versions,
+                            )
                         )
+                        for subdir in extract_subdirs_to_cache()
+                    ]
+                    for f in futures:
```
Comment (dholth) [inline conda_index/index/__init__.py:564]: clean up before merge

Comment (dholth) [reply conda_index/index/__init__.py:564]: I thought the with statement: would wait for all the submitted futures, but it appears we have to call .result() on each.

Comment (barabo) [reply conda_index/index/__init__.py:564]: Not crazy about having important side effects happen inside logging statements.

```suggestion
                    for f in futures:
                        result = f.result()
                        log.debug(f"From subprocess {result=}")
```

**Review by barabo — APPROVED** [review]
Awesome how much easier it is to read without the `tqdm`.  Only one minor suggestion to consider!

**Pull request merged**